modules/services/xidlehook.nix

```
@@ -0,0 +1,144 @@
+# Wrapper around xidlehook.
+#
+# This module has been created by 'berbiche' and modified by 'dschrempf'.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.xidlehook;
+
+  notEmpty = list: filter (x: x != "" && x != null) (flatten list);
+
+  timers =
+    let
+      toTimer = timer:
+        "--timer ${toString timer.delay} ${escapeShellArgs [ timer.command timer.canceller ]}";
+    in map toTimer (filter (timer: timer.command != null) cfg.timers);
+
+  script = pkgs.writeShellScript "xidlehook" ''
+    ${concatStringsSep "\n"
+      (mapAttrsToList (name: value: "export ${name}=${value}") cfg.environment or {})
+     }
+    ${concatStringsSep " " (notEmpty [
+      "${cfg.package}/bin/xidlehook"
+      (optionalString cfg.once "--once")
+      (optionalString cfg.not-when-fullscreen "--not-when-fullscreen")
+      (optionalString cfg.not-when-audio "--not-when-audio")
+      timers
+    ])}
+  '';
+in
+{
+  meta.maintainers = [ maintainers.dschrempf ];
+
+  options.services.xidlehook = {
+    enable = mkEnableOption "xidlehook systemd service";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.xidlehook;
+      defaultText = "pkgs.xidlehook";
+      description = "The package to use for xidlehook.";
+    };
+
+    environment = mkOption {
+      type = types.attrsOf types.str;
+      default = { };
+      example = literalExample ''
+        {
+          "primary-display" = "$(xrandr | awk '/ primary/{print $1}')";
+        }
+      '';
+      description = ''
+        Extra environment variables to be exported in the script.
+        These options are passed unescaped as <code>export NAME=value</code>.
+      '';
+    };
+
+    not-when-fullscreen = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Disable locking when a fullscreen application is in use.";
+    };
+
+    not-when-audio = mkOption {
+      type = types.bool;
+      default = false;
+      example = true;
+      description = "Disable locking when audio is playing.";
+    };
+
+    once = mkEnableOption "running the program once and exiting";
+
+    timers = mkOption {
+      type = types.listOf (types.submodule {
+        options = {
+          delay = mkOption {
+            type = types.ints.unsigned;
+            example = 60;
+            description = "Time before executing the command.";
+          };
+          command = mkOption {
+            type = types.nullOr types.str;
+            example = literalExample ''
+              ''${pkgs.libnotify}/bin/notify-send "Idle" "Sleeping in 1 minute"
+            '';
+            description = "Command executed after the idle timeout is reached.";
```

```suggestion
            description = ''
              Command executed after the idle timeout is reached.
              Path to executables are accepted.
              The command is automatically escaped.
            '';
```
Feel free to rephrase.